tests/platform/test_link_flap.py

```diff
@@ -0,0 +1,80 @@
+import logging
+
+import pytest
+
+from common.platform.device_utils import fanout_switch_port_lookup
+from common.utilities import wait_until
+
+class TestLinkFlap:
+    def __init__(self):
+        self.ports_shutdown_by_test = set()
+
+
+    def __get_dut_if_facts(self, dut):
+        interface_facts = dut.interface_facts()
+        ansible_facts   = interface_facts['ansible_facts']
+        if_facts        = ansible_facts['ansible_interface_facts']
+
+        return if_facts
+
+
+    def __check_if_status(self, dut, dut_port, exp_state):
+        ifstate = self.__get_dut_if_facts(dut)[dut_port]
+        return ifstate['active'] == exp_state
+
+
+    def __toggle_one_link(self, dut, dut_port, fanout, fanout_port):
+        logging.info("Testing link flap on {}".format(dut_port))
+
+        ifstate = self.__get_dut_if_facts(dut)[dut_port]
+        assert ifstate['active'], "dut port {} is down".format(dut_port)
```

Comment:
There is an issue with this assertion. Some device has ports that are not enabled. Looking into getting admin state.

Reply:
Changed to use show_interface to obtain interface status.